pkg/didcomm/dispatcher/api.go

```diff
@@ -23,9 +23,10 @@ type DIDCommMsg struct {
 	// Outbound indicates the direction of this DIDComm message:
 	//   - outgoing (to another agent)
 	//   - incoming (from another agent)
-	Outbound bool
-	Type     string
-	Payload  []byte
+	Outbound            bool
+	Type                string
+	Payload             []byte
+	OutboundDestination *Destination
```

Comment:
Just wondering if we need to add `OutboundDestination` to `DIDCommMsg`. I see in this PR, it's just wrapped inside `DIDCommMsg` to pass around. I feel we can explicitly pass it as function arguments than adding it here as `DIDCommMsg` is used to pass messages through handlers/eventing.

@llorllale @troyronda @fqutishat thoughts?

Reply:
I was nt even sure where to add this outbound destination, The idea is service needs to know this destination before it starts executing the state actions. 

Which function argument you are recommending ?

Reply:
I think we should be persisting the keys and destination in the store and use from there.

Reply:
Can you please add a TODO if we are planning to refactor the OutboundDestination ?